crowbar_framework/app/models/service_object.rb

```diff
@@ -1169,7 +1171,14 @@ def apply_role(role, inst, in_queue, bootstrap = false)
         end
       end # roles.each
 
-      batches << [roles, nodes_in_batch] unless nodes_in_batch.empty?
+      if pnodes_in_batch == nodes_in_batch && !proles.nil?
```

Comment:
```element_order = [
[b1]
[c1,c2]
]
batch 1: node1=[b1] node2=[b1]
batch 2: node1=[c1]
```
It think : this could break, if `c1` needs `b1` to be entirely functional

Reply:
> this could break, if c1 needs b1 to be entirely functional

Not necessarily, but that demonstrates my point 2) below - it depends on whether all the recipes for b1 were ordered to happen before the recipes for c1.  For example:

```
element_order = [
[b1]
[c1,c2]
]
batch 1: node1=[b1] node2=[b1]
batch 2: node1=[c1] node2=[c1]
```

This PR would change that to:

```
batch 1: node1=[b1,c1] node2=[b1,c1]
```

If some of c1's recipes come before b1's in the runlist then there's a problem.  Normally runlist ordering is determined by `chef_order`, but this can be overridden per role by `element_run_list_order`.  The only place we actually use that is with pacemaker remotes, but even there I'm not sure we actually use it - maybe the code only has remnants of a previous usage which we reverted when @vuntz came up with a different approach?  I can't remember but @vuntz will know for sure.

Reply:
> If some of c1's recipes come before b1's in the runlist then there's a problem. 
yes this is correct.

I did look at `element_run_list_order` in nova, ceilometer swift and other barclamps. 
Didnt find the issue you mentioned. As each one of them had explicit priorities.

Reply:
Yeah, I think I must have done the grep wrong last time because I see them now.